vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -393,16 +371,16 @@ def prepare_model_input(
                     self.input_batch.soft_remove_request(req_id)
                 self.input_batch.refresh_sampling_metadata()
 
-            (input_tokens, input_positions, input_masks) = \
-                self._prepare_decode(scheduler_output.scheduled_cached_reqs)
-
-        sampling_metadata = self.input_batch.sampling_metadata
+            (input_tokens, input_positions,
+             input_masks) = self._prepare_decode(
```

Comment:
If we wanted to push the refactor just a tiny bit further, both `_prepare_prompt` and `_prepare_decode` can now return a valid `ModelForwardInputs`

Reply:
`_prepare_prompt` returns `seq_lens` as well, not sure if we want to add that as a field within the `ModelForwardInputs`

Reply:
Although I don't see `seq_lens` being used anywhere in CB, probably safe to delete it?

Reply:
probably! I assume that's a holdover from the V0 times